simulator/simulator.cpp

```diff
@@ -38,8 +39,16 @@ class SimulatorFactory {
     template<typename T, Endian e>
     struct TBuilder : public Builder {
         TBuilder() = default;
-        std::unique_ptr<Simulator> get_funcsim( bool log) final { return std::make_unique<FuncSim<T>>( e, log); }
-        std::unique_ptr<CycleAccurateSimulator> get_perfsim() final { return std::make_unique<PerfSim<T>>( e); }
+        std::unique_ptr<Simulator> get_funcsim( bool log) final {
+            auto func_sim = std::make_unique<FuncSim<T>>( e, log);
+            func_sim->set_isa( Builder::isa);
+            return func_sim;
+        }
+        std::unique_ptr<CycleAccurateSimulator> get_perfsim() final {
+            auto perf_sim = std::make_unique<PerfSim<T>>( e);
+            perf_sim->set_isa( Builder::isa);
```

Comment:
It's better to use `this->isa` so there would be less changes if base class is renamed.